### PR TITLE
Add enable_docker_enroot_pyxis, enable_observability in config.py

### PR DESCRIPTION
**Pull request by shimomut**
*Description of changes:*

* Adding enable_docker_enroot_pyxis, enable_observability in config.py so that lifecycle_config.py doesn't have to be edited to enable/disable features.

* Install docker/enroot/pyxis by default.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: 1.architectures/5.sagemaker-hyperpod/LifecycleScripts/base-config/lifecycle_script.py [hunk 1]
```diff
@@ -182,20 +182,21 @@ def main(args):
...
-        #     ExecuteBashScript("./utils/install_slurm_exporter.sh").run()
-        #     ExecuteBashScript("./utils/install_prometheus.sh").run()
+        # Install Docker/Enroot/Pyxis
+        if Config.enable_docker_enroot_pyxis:
+            ExecuteBashScript("./utils/install_docker.sh").run()
+            ExecuteBashScript("./utils/install_enroot_pyxis.sh").run(node_type)
+
+        # Install metric exporting software and Prometheus for observability
+        if Config.enable_observability:
+            if node_type == SlurmNodeType.COMPUTE_NODE:
```
Comment (sean-smith) [inline 1.architectures/5.sagemaker-hyperpod/LifecycleScripts/base-config/lifecycle_script.py:192]: Let's include `ExecuteBashScript("./utils/install_docker.sh").run()`

File: 1.architectures/5.sagemaker-hyperpod/LifecycleScripts/base-config/lifecycle_script.py [hunk 2]
```diff
@@ -182,20 +182,21 @@ def main(args):
...
+                ExecuteBashScript("./utils/install_dcgm_exporter.sh").run()
+                ExecuteBashScript("./utils/install_efa_node_exporter.sh").run()
+
+            if node_type == SlurmNodeType.HEAD_NODE:
+                wait_for_scontrol()
+                ExecuteBashScript("./utils/install_slurm_exporter.sh").run()
```
Comment (sean-smith) [inline 1.architectures/5.sagemaker-hyperpod/LifecycleScripts/base-config/lifecycle_script.py:198]: same here

File: 1.architectures/5.sagemaker-hyperpod/LifecycleScripts/base-config/config.py [hunk 3]
```diff
@@ -2,6 +2,15 @@
 # Basic configuration parameters
 class Config:
 
+    # Set true if you want to install Docker/Enroot/Pyxis.
+    enable_docker_enroot_pyxis = True
+
+    # Set true if you want to install 
+    # DCGM Exporter and EFA Node Exporter are installed on compute nodes, 
+    # Slurm Exporter and Prometheus are installed on controller node.
+    # Note: enable_docker_enroot_pyxis must be set to True.
```
Comment (sean-smith) [inline 1.architectures/5.sagemaker-hyperpod/LifecycleScripts/base-config/config.py:11]: Let's remove this note and just add it

**Comment by shimomut** [comment]
Addressed feedback. Please review again.

File: 1.architectures/5.sagemaker-hyperpod/LifecycleScripts/base-config/lifecycle_script.py (hunk 1 above)
Comment (shimomut) [reply 1.architectures/5.sagemaker-hyperpod/LifecycleScripts/base-config/lifecycle_script.py:192]: Done

File: 1.architectures/5.sagemaker-hyperpod/LifecycleScripts/base-config/lifecycle_script.py (hunk 2 above)
Comment (shimomut) [reply 1.architectures/5.sagemaker-hyperpod/LifecycleScripts/base-config/lifecycle_script.py:200]: Done

File: 1.architectures/5.sagemaker-hyperpod/LifecycleScripts/base-config/config.py (hunk 3 above)
Comment (shimomut) [reply 1.architectures/5.sagemaker-hyperpod/LifecycleScripts/base-config/config.py]: Done

**Comment by shimomut** [comment]
If I run install_docker.sh twice, it fails. Investigating now.

**Comment by shimomut** [comment]
Fixed docker 2nd installation failure

**Review by sean-smith — APPROVED** [review]

**Comment by sean-smith** [comment]
LGTM - let's make sure to update the workshop and merge at the same time.

**Pull request merged**